### ISSUE TITLE
[LayoutLM] How to extract the best candidate after token classification?

### ISSUE THREAD

**Issue by renjithsasidharan** [microsoft/unilm#139]
Let's assume the model predicts the following for an input sequence.
```
The O            
creation. O
date O
is O
27 B-DATE
Aug I-DATE
2020 I-DATE
and  O
update. O
date.  O
is  O
01-09-2020 B-DATE
```

How do you pick the best candidate for **creation date** from logist values?

**Comment by wolfshow** [comment]
@renjithsasidharan This is not related to LayoutLM training/fine-tuning. It depends on your downstream tasks and how the data is labeled. Typically, the downstream tasks should have more fine-grained labels instead of only "B-Date" and "I-Date".

**Comment by renjithsasidharan** [comment]
But how would you do it? Can I average/multiply softmax scores of (B-DATE and I-DATE) and pick the best one?

**Comment by wolfshow** [comment]
@renjithsasidharan Yes, averaing the softmax scores might be a straightforward way. In addition, if the sequence contains more than one entity for the same field, I think you may need to consider voting or other methods which helps to pick the best one.

**Comment by renjithsasidharan** [comment]
Thanks for the info, I will close this ticket.
